backend/src/Designer/Models/FooterFile.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+public class FooterFile
+{
+    [JsonPropertyName("$schema")]
+    public string Schema { get; set; }
+
+    [JsonPropertyName("footer")]
+    public List<FooterLayout> Footer { get; set; }
+}
+
+public class FooterLayout
+{
+    [JsonPropertyName("type")]
+    public string Type { get; set; }
+
+    [JsonPropertyName("title")]
+    public string Title { get; set; }
+
+    [JsonPropertyName("target")]
+    public string Target { get; set; }
+
+    [JsonPropertyName("icon")]
+    public string Icon { get; set; }
+}
+
+public enum FooterComponentType
+{
+    Email,
+    Link,
+    Phone,
+    Text
+}
+
+public enum IconType
+{
+    Information,
+    Email,
+    Phone
+}
+
+// Specific component classes
+public class EmailComponent : FooterFile
```

Comment:
Are these component classes used anywhere, or just defined here?
C# apparently does not support union types in the same way as typescript. In TS, you would just say
```js
type ComponentType = EmailComponent | LinkComponent | PhoneComponent | TextComponent;
```
In C# you would have to use records to do something similar, but it's not quite the same.

Perhaps we should just expose the basic `FooterLayout` class, but make the `Target` and `Icon` properties nullable? And then leave out the component specific classes. They don't seem to be in use anyways. That would probably be good enough, although it does not define explicitly which properties each component type can have. I think implementing a complete definition here would be quite complicated 🤔 🙈